pkg/kubelet/kuberuntime/kuberuntime_container.go

```diff
@@ -77,6 +79,26 @@ func (m *kubeGenericRuntimeManager) startContainer(podSandboxID string, podSandb
 		m.recorder.Eventf(ref, v1.EventTypeWarning, events.FailedToCreateContainer, "Failed to create container with error: %v", err)
 		return "Generate Container Config Failed", err
 	}
+
+	// Delegate to registered isolators.
+	eventReply, err := containermanager.GetEventDispatcherSingleton().PreStartContainer(podSandboxConfig.Metadata.Name, containerConfig.Metadata.Name)
+	if err != nil {
+		return "", fmt.Errorf("failed to collect isolator responses for container %q: %v", containerConfig.Metadata.Name, err)
+	}
+	if eventReply.Error != "" {
+		return "", fmt.Errorf("isolator returned error: %s", eventReply.Error)
+	}
+	// Apply aggregated isolator responses to the container create config.
+	for _, control := range eventReply.IsolationControls {
```

Comment:
@ConnorDoyle I wonder if we can move that logic somewhere inside eventDispatcher interface.
In future when there will be a need to add i.e modification of env variables as in #50 the code changes will be limited hopefully only (obviously that depends on what the isolator does) to event_dispatcher.go file thus making it easier and less "impacting" on core kubelet code. Example `ResourceConfigFromReplies()` function.
What are your thoughts ?